### PR TITLE
ADD: isTorCapable

### PR DESCRIPTION
**Pull request by marcosrdz**
Removes the need to manual code adjustments on platforms that aren't compatible with react-native-tor

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/0w1pjrffqyrefth35522366w7m?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/7334)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/g6v2nhxgx8d7qvry52btx0kunr?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/7335)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/b0e33m8r3uzcrb817ztdypupbr?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/7337)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/wv4khz6kkdh4h3hvn47b0hbnec?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/7338)

File: blue_modules/BlueElectrum.js [hunk 1]
```diff
@@ -101,13 +102,15 @@ async function connectMain() {
...
-    mainClient = new ElectrumClient(
-      usingPeer.host.endsWith('.onion') ? torrific : global.net,
-      global.tls,
-      usingPeer.ssl || usingPeer.tcp,
-      usingPeer.host,
-      usingPeer.ssl ? 'tls' : 'tcp',
-    );
+    mainClient = isTorCapable
+      ? new ElectrumClient(
+          usingPeer.host.endsWith('.onion') ? torrific : global.net,
```
Comment (Overtorment) [inline blue_modules/BlueElectrum.js:107]: i think that can be reduced to

```
(usingPeer.host.endsWith('.onion') && isTorCapable) ? torrific : global.net,
```

File: blue_modules/BlueElectrum.js [hunk 2]
```diff
@@ -5,11 +5,12 @@ import { LegacyWallet, SegwitBech32Wallet, SegwitP2SHWallet } from '../class';
 import DefaultPreference from 'react-native-default-preference';
 import RNWidgetCenter from 'react-native-widget-center';
 import loc from '../loc';
+import { isTorCapable } from '../blue_modules/environment';
```
Comment (Overtorment) [inline blue_modules/BlueElectrum.js:8]: you corrected `require('./torrific');` below but added extra `../blue_modules/` here kek

File: blue_modules/BlueElectrum.js [hunk 3]
```diff
@@ -805,13 +808,9 @@ module.exports.calculateBlockTime = function (height) {
...
 module.exports.testConnection = async function (host, tcpPort, sslPort) {
-  const client = new ElectrumClient(
-    host.endsWith('.onion') ? torrific : global.net,
-    global.tls,
-    sslPort || tcpPort,
-    host,
-    sslPort ? 'tls' : 'tcp',
-  );
+  const client = isTorCapable
+    ? new ElectrumClient(host.endsWith('.onion') ? torrific : global.net, global.tls, sslPort || tcpPort, host, sslPort ? 'tls' : 'tcp')
```
Comment (Overtorment) [inline blue_modules/BlueElectrum.js:812]: looks like it can also be simplified to reduce diff

File: screen/settings/NetworkSettings.js [hunk 4]
```diff
@@ -35,7 +35,7 @@ const NetworkSettings = () => {
             chevron
           />
         )}
-        {!isDesktop && <BlueListItem title={loc.settings.tor_settings} onPress={navigateToTorSettings} testID="TorSettings" chevron />}
+        {isTorCapable && <BlueListItem title={loc.settings.tor_settings} onPress={navigateToTorSettings} testID="TorSettings" chevron />}
```
Comment (Overtorment) [inline screen/settings/NetworkSettings.js:38]: i have a feeling that might crash with cryptic "cant render text outside of <Text> tag"

**Review by Overtorment — CHANGES_REQUESTED** [review]

File: blue_modules/BlueElectrum.js (hunk 2 above)
Comment (marcosrdz) [reply blue_modules/BlueElectrum.js:8]: lol silly auto import

File: screen/settings/NetworkSettings.js (hunk 4 above)
Comment (marcosrdz) [reply screen/settings/NetworkSettings.js:38]: it doesnt.

File: blue_modules/BlueElectrum.js (hunk 1 above)
Comment (marcosrdz) [reply blue_modules/BlueElectrum.js]: fixed

File: blue_modules/BlueElectrum.js (hunk 3 above)
Comment (marcosrdz) [reply blue_modules/BlueElectrum.js]: fixed.

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/bt9d4r8h0epaftae838qdz3zf4?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/7344)

**Review by Overtorment — CHANGES_REQUESTED** [review]
tests fail.
also rebasing would not hurt, so a bunch of WIP commits wont appear in our git history

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/urda4je389q8e8d2waeht3an40?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/7346)

**Review by Overtorment — APPROVED** [review]
THANKS KФMЯAD KЯAKЕИ

**Comment by GladosBlueWallet** [comment]
Unbelievable. You, [subject name here], must be the pride of [subject hometown here]!

**Pull request merged**